prometheusbp/promtest/doc.go

```diff
@@ -0,0 +1,2 @@
+// Package promtest provides test utilites for testing Prometheus metrics.
+package promtest
```

Comment:
🤷‍♀️ I don't feel strongly either way.

Reply:
Thinking about this more, I do want `[prometheusbp.DefaultBuckets](https://github.com/reddit/baseplate.go/blob/prometheus/prometheusbp/metrics.go#L8)` to be useable by services using baseplate.go as well as the `promtest` stuff.  So maybe it would be better to leave where it is. Thoughts @kylelemons

Reply:
I think you want a few packages here, not just one:

* `prometheusbp` -- general-purpose code for use anywhere (e.g. `DefaultBuckets`)
* `prometheusbp/promtest` -- general-purpose test helpers for anyone to use (e.g. the `CheckDelta` stuff)
* `internal/prometheusbp/spectest` -- specific test helper(s) to unify baseplate spec prom compliance logic between other baseplate.go packages